### PR TITLE
Move src/qgsquick to src/core

### PR DESCRIPTION
**Pull request by nirvn**
@m-kuhn , as discussed over google chat.

### CODE REVIEW

File: src/core/featurelistextentcontroller.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 #include "multifeaturelistmodel.h"
 #include "featurelistmodelselection.h"
-#include "qgsquickmapsettings.h"
+#include "qgsquick/qgsquickmapsettings.h"
```
Comment (m-kuhn) [inline src/core/featurelistextentcontroller.h:23]: Can we just add the previx to the include paths in CMakeLists.txt instead?

**Comment by qfield-fairy** [comment]
Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-b7bb03a9e8e1df1b78693da9af975a45441ad6b9-arm64_v8a.apk)

Other architectures:

  - [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-b7bb03a9e8e1df1b78693da9af975a45441ad6b9-armv7.apk)
  - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-b7bb03a9e8e1df1b78693da9af975a45441ad6b9-x86_64.apk)
  - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-b7bb03a9e8e1df1b78693da9af975a45441ad6b9-x86.apk)

**Comment by qfield-fairy** [comment]
Ta-daaa, freshly created APKs are available:
  - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-1e4579a19ae3a1d87304997506df7ccb861225cb-arm64_v8a.apk)

Other architectures:

  - [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-1e4579a19ae3a1d87304997506df7ccb861225cb-armv7.apk)
  - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-1e4579a19ae3a1d87304997506df7ccb861225cb-x86_64.apk)
  - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1396-1e4579a19ae3a1d87304997506df7ccb861225cb-x86.apk)

**Comment by nirvn** [comment]
@m-kuhn , done.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**